components/brave_rewards/resources/creator_detection/twitch.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { CreatorInfo, initializeDetector } from './creator_detection'
+import { log, getPathComponents, pollFor } from './helpers'
+
+initializeDetector(() => {
+
+  function scrapeIcon() {
+    const elem =
+      document.querySelector('.channel-info-content .tw-avatar [src]')
+    return (elem && elem.getAttribute('src')) || ''
+  }
+
+  function getChannelTitleElement() {
+    return document.querySelector('h1.tw-title')
+  }
+
+  function scrapeChannelName() {
+    const elem = getChannelTitleElement()
+    return elem && elem.textContent || ''
+  }
+
+  function firstPathComponent(path: string) {
+    return getPathComponents(path)[0] || ''
+  }
+
+  function isVideoPath() {
+    return firstPathComponent(location.pathname) === 'videos'
+  }
+
+  const excludedPaths = new Set([
```

Comment:
do we have a plan for keeping this up to date?

Reply:
Not exactly. Again, this list is ported from the existing Greaselion scripts. I'm hoping that we can make improvements in a follow-up as needed.